controllers/rhmi/bootstrapReconciler.go

```diff
@@ -762,6 +770,43 @@ func (r *Reconciler) processQuota(installation *rhmiv1alpha1.RHMI, namespace str
 	return nil
 }
 
+func (r *Reconciler) reconcileCustomSMTP(ctx context.Context, serverClient k8sclient.Client) (rhmiv1alpha1.StatusPhase, error) {
+
+	smtp, err := cs.GetCustomAddonValues(serverClient, r.installation.Namespace)
+	if err != nil {
+		return integreatlyv1alpha1.PhaseFailed, err
+	}
+
+	validation := cs.ParameterValidation(smtp)
+	if validation == cs.Valid {
+		phase, err := cs.CreateOrUpdateCustomSMTPSecret(ctx, serverClient, smtp, r.installation.Namespace)
+		if err != nil {
+			return phase, err
+		}
+		r.installation.Status.CustomSmtp = &rhmiv1alpha1.CustomSmtpStatus{}
+		r.installation.Status.CustomSmtp.Active = true
+
+	} else if validation == cs.Partial {
+		phase, err := cs.DeleteCustomSMTP(ctx, serverClient, r.installation.Namespace)
+		if err != nil {
+			return phase, err
+		}
+
+		errorString := cs.ParameterErrors(smtp)
+		r.installation.Status.CustomSmtp.Active = false
+		r.installation.Status.CustomSmtp.Error = fmt.Sprintf("Custom SMTP partial configured, missing fields: %s", errorString)
+
+	} else {
```

Comment:
I would rather make this an if on the validation rather than an else in case the validation return gets updated in the future. 
After the if you could have a return of error of unknown validation type.

Reply:
A switch statement would make the logic more readable here so i'm in favour of that. 👍

Reply:
While change. The default will be to throw an error if there is an unknown validation type.